packages/core/lib/modules/accounts.js

```diff
@@ -70,36 +64,37 @@ export class Accounts {
     return balance;
   }
 
-  async getBalance(accountId) {
-    const currency = await this.findCorrelationByType(accountId, 'Currency');
+  async getBalance(address, skipFetch?) {
+    if (!skipFetch) {
+      await this.fetchBalance(address);
+    }
+
+    const currency = await this.findCorrelationByType(address, 'Currency');
```

Comment:
what are the chances of failure in this call? Should we wrap it and handle it?

Reply:
I added some validation here. it should throw an error if the currency object is not defined, in that case something is wrong with the state or the implementation, so its good to throw instead of propagating the error.